### DISCUSSION TITLE
Receiving duplicated messages after client reconnection to broker.

### DISCUSSION THREAD

**Discussion by andrebeltrami** [davidepianca98/KMQTT#15] [Q&A]
Hi Davide and all,
I'm testing this broker implementation with android MQTT clients and I would like to know if this behavior is some implementation decision in the broker side. 
Using a broker such as EMQX and mosquitto for example, when an android MQTT client disconnect and connect again after a while, the client can receive only the missing messages during the disconnection period (clients were set with QoS=1 and cleanSession=False, unique clientID). However, with same client configuration but using this Broker implementation after an android MQTT reconnection, the client receives all messages even messages that it already received before. There is a way to configure in the broker to after a reconnection sent only the missing events instead of all events?

Thank you. Regards, André.

**Comment by davidepianca98** [comment]
Hello, thank you for letting me know. Can you please tell me which MQTT version are you using? Also are you sending retained messages?

**Reply by andrebeltrami** [reply]
The MQTT version is 3.1.1 and the messages are published with retained = false.

**Reply by davidepianca98** [reply]
When you get the additional packets, do they have the dup flag set to indicate they are being sent again? With QoS 1 duplicate packets can occur if the broker didn't receive the acknowledge message from the client. I would like to know what do you mean with "the client receives all messages even messages that it already received before", because if the packets have been received by the client but the broker didn't receive the ack because the client disconnected, then it's going to get sent again. On the other hand if you mean older packets that you are sure have been acknowledged there is something wrong with the implementation.

**Reply by andrebeltrami** [reply]
Hi Davide, you are right. I tested it again and worked fine, probably the broker didn't received the ACK from those clients. Thank you for the help.
